server/gitrest/packages/gitrest-base/src/routes/index.ts

```diff
@@ -36,22 +36,26 @@ export interface IRoutes {
 
 export function create(
 	store: nconf.Provider,
-	fileSystemManagerFactory: IFileSystemManagerFactory,
+	fileSystemManagerFactories: IFileSystemManagerFactories,
```

Comment:
Maybe this is too much work but at first I got confused with this new `IFileSystemManagerFactories` and how it relates to `IFileSystemManagerFactory`... The first though that came to mind was that this was a collection of instances of `IFileSystemManagerFactory`, but I now see that this works as a factory of `IFileSystemManagerFactory`, or, in other words, a `IFileSystemManagerFactoryFactory`. That name would be very ugly and equally as confusing, perhaps more, but I was wondering if we can rearrange or rename things in a way that becomes easier to understand. For example, instead of having a factory for IFileSystemManagerFactoryFactory, could we instead just have a single level/layer factory which now takes into account the new ephemeral container parameter?

For example, now that we have 3 implementations of `IFileSystemManagerFactory` in https://github.com/microsoft/FluidFramework/blob/244737ea69a4057cf08f12b988bbdb4b9478c2f5/server/gitrest/packages/gitrest-base/src/utils/filesystems.ts#L17, could we merge those implementations into a single one, and have the `create()` implementation basically do what `getFileSystemManagerFactories()` is doing?

Reply:
The way I was thinking about this was that we now have two IFileSystemManagerFactories (durable and ephemeral) and based on the ephemeral flag one of them will be chosen. These two seemed like a part of one logical unit which is why the two level organization made sense to me. In the future, if we want to replace redis as a ephemeral store with something like cosmos db, this will be cleaner as opposed to having all the selection logic in the single layer create() where not all the IFileSystemManagerFactory are related to the ephemeral flag. I am currently inclined to leave it this way. Please let me know if you feel strongly that the single layer factory seems better.